### ISSUE TITLE
False positive on `Style/HashEachMethods` cop with block parameter destructuring

### ISSUE THREAD

**Issue by swordray** [rubocop/rubocop#12441]
```ruby
changes = { name: ["bill", "bob"] }
changes.each { |key, (_, value)| puts key, value }
```

## Expected behavior

no offenses

## Actual behavior

```ruby
Offenses:

each.rb:2:1: C: [Correctable] Style/HashEachMethods: Use each_key instead of each and remove the unused (_, value) block argument.
changes.each { |key, (_, value)| puts key, value }
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

`rubocop --debug --only Style/HashEachMethods`

## RuboCop version

```
$  rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin23]
```
